### PR TITLE
Feature us5 - Get Detail / List activities

### PR DESCRIPTION
**Pull request by Pia0912**
US5  Get Detail  & List activities

Backend:
- add Activity class, add Activities to Hobby class (therefore edit other classes & tests)
- add GET Details Controller Endpoint
- add List Activities Controller Endpoint
- add Exception Handling
- cleanup code

Frontend:
- add Activity Model
- add ActivityItems & DetailPage
- add routes for endpoints
- add Star Rating

### CODE REVIEW

File: backend/src/main/java/de/neuefische/capstone/pia/backend/exceptions/NoSuchActivityException.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus(code = HttpStatus.NOT_FOUND, reason = "Activity not found!")
+public class NoSuchActivityException extends RuntimeException {
+    public NoSuchActivityException(String id) {
+        super("Activity not found for ID: " + id);
+    }
+}
```
Comment (cananozyol) [inline backend/src/main/java/de/neuefische/capstone/pia/backend/exceptions/NoSuchActivityException.java:11]: 👍

File: frontend/src/components/HobbyDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+export default function HobbyDetail() {
+    const [hobby, setHobby] = useState<Hobby | null>(null);
+    const [activities, setActivities] = useState<Activity[] | null>(null);
+    const params = useParams();
+    const navigate = useNavigate();
+
+    const location = useLocation();
+    const selectedColor = location.state?.selectedColor || "#f2f2f2";
+
+    useEffect(() => {
```
Comment (cananozyol) [inline frontend/src/components/HobbyDetail.tsx:18]: maybe put that in your custom hook 🤙🏻

**Review by cananozyol — APPROVED** [review]
Nice job Pia, especially your exceptions!! LGTM 🚀🚀🚀

File: frontend/src/components/HobbyDetail.tsx (hunk 2 above)
Comment (Pia0912) [reply frontend/src/components/HobbyDetail.tsx:18]: Thanks for the great conversation before - now that makes more sense to me! 👍🏻

**Pull request merged**